### PR TITLE
Add `evaluate` extra to specify extra dependencies required for `mlflow.evaluate`

### PR DESCRIPTION
**Pull request by harupy**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Add `evaluate` extra to specify extra dependencies required for `mlflow.evaluate`.

## How is this patch tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests (describe details, including test results, below)

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/gateway`: AI Gateway service, Gateway client APIs, third-party Gateway integrations
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/models/evaluation/base.py [hunk 1]
```diff
@@ -1262,6 +1262,15 @@ def evaluate(
...
           explicitly set to ``True``.
 
+    .. note::
+
+        To compute and log a full set of metrics and artifacts supported by the default
+        evaluator, use the following command to install required extra dependencies:
+
+        .. code-block:: bash
+
+            pip install 'mlflow[evaluate]'
```
Comment (harupy) [inline mlflow/models/evaluation/base.py:1272]: I think this is better than `question-answering requires a, b, and c. text-summarization requires x, y, and z`.

Comment (chenmoneygithub) [reply mlflow/models/evaluation/base.py:1272]: Agreed! The current solution is hard for users to follow

File: setup.py [hunk 2]
```diff
@@ -171,6 +171,17 @@ def run(self):
             "boto3>1",
         ],
         "gateway": GATEWAY_REQUIREMENTS,
+        # Extra dependencies required for `mlflow.evaluate`
```
Comment (chenmoneygithub) [inline setup.py:174]: nit: period at the end

File: .github/workflows/master.yml [hunk 3]
```diff
@@ -286,7 +286,7 @@ jobs:
       - name: Install dependencies
         run: |
           source ./dev/install-common-deps.sh
-          pip install pyspark "numpy<1.24.0" torch transformers
+          pip install '.[evaluate]' pyspark
```
Comment (chenmoneygithub) [inline .github/workflows/master.yml:289]: Nonblocking just curious - why is `pyspark` required for `mlflow.evaluate`?

**Review by chenmoneygithub — APPROVED** [review]
Thanks Haru! LGTM, some minor comments

**Comment by harupy** [comment]
We decided not to to this.

**Pull request closed without merging**